src/app.ts

```diff
@@ -85,8 +85,21 @@ export class App {
     this.eventAggregator.subscribe("Network.wrongNetwork", async (info: { provider: any, connectedTo: string, need: string }) => {
 
       let notChanged = true;
+
+      let message = `<p>Your wallet is connected to ${info.connectedTo ?? "an unknown network"}, but to interact with deals we need you to connect to ${info.need}.  Do you want to switch your connection ${info.need} now?<p>`;
+      let header: string | undefined = undefined;
+      if (await this.ethereumService.isSafeApp()) {
```

Comment:
can this return an error? if so, do we need to handle it somehow ?

Reply:
good point, I added try/catch es to all relevant methods

Reply:
[0a983c3](https://github.com/PrimeDAO/prime-deals-dapp/pull/1119/commits/0a983c3b6cf95ecf1e0adb1bdf94653fb385190f)